src/components/ProductCard/ProductCard.jsx

```diff
@@ -1,24 +1,44 @@
 import React from "react";
 import "./ProductCard.css";
-import { Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
+import { addToCartHandler, checkAction } from "../../functions/";
+import { useAuth, useCart } from "../../context/";
+
+const ProductCard = ({ product }) => {
+  const {
+    _id: id,
+    imageSrc: image,
+    title,
+    badge,
+    price,
+    discount,
+    inStock,
+    sizes,
+  } = product;
+
+  const navigate = useNavigate();
+  const size = sizes[0];
+
+  const {
+    authState: { token },
+  } = useAuth();
+  const {
+    cartState: { cart },
+    cartDispatch,
+  } = useCart();
 
-const ProductCard = ({
-  id,
-  image,
-  title,
-  alt,
-  badge,
-  price,
-  discount,
-  inStock,
-}) => {
   const priceBefore =
     Number(price) +
```

Comment:
if you are using string concat here using "+" then I'll suggest using string literal as it is good practice

Reply:
Actually, It's not concatenation.. I am suming up two values

Reply:
ok np i got confused with two + + sign that's why i added.